client/sites/components/sites-dataviews/index.tsx

```diff
@@ -185,47 +198,31 @@ const DotcomSitesDataViews = ( {
 				getValue: () => null,
 			},
 		],
-		[ __, openSitePreviewPane, userId, dataViewsState, setDataViewsState, siteStatusGroups ]
+		[ __, openSitePreviewPane, userId, siteStatusGroups ]
 	);
 
-	const siteSearchLabel = __( 'Search sites…' );
-
-	// Create the itemData packet state
-	const [ itemsData, setItemsData ] = useState< ItemsDataViewsType< SiteExcerptData > >( {
-		items: sites,
-		itemFieldId: 'ID',
-		searchLabel: siteSearchLabel,
-		fields,
-		actions: [],
-		setDataViewsState: setDataViewsState,
-		dataViewsState: dataViewsState,
-		onSelectionChange,
-		pagination: paginationInfo,
-		defaultLayouts: { table: {} },
-	} );
-
-	// Update the itemData packet
-	useEffect( () => {
-		setItemsData( ( prevState: ItemsDataViewsType< SiteExcerptData > ) => ( {
-			...prevState,
-			items: sites,
-			fields,
-			// actions: actions,
-			setDataViewsState,
-			dataViewsState,
-			searchLabel: siteSearchLabel,
-			selectedItem: dataViewsState.selectedItem,
-			selection: getSelection( dataViewsState ),
-			pagination: paginationInfo,
-		} ) );
-	}, [ fields, dataViewsState, paginationInfo, setDataViewsState, sites, siteSearchLabel ] ); // add actions when implemented
-
 	return (
-		<ItemsDataViews
-			data={ itemsData }
-			isLoading={ isLoading }
-			className="sites-overview__content"
-		/>
+		<div className="sites-overview__content">
+			<DataViews
+				data={ sites }
+				fields={ fields }
+				onChangeView={ ( newView: View ) => setDataViewsState( () => newView ) }
+				view={ dataViewsState }
+				search
+				searchLabel={ __( 'Search sites…' ) }
+				selection={ getSelection() }
+				paginationInfo={ paginationInfo }
+				getItemId={ ( item ) => {
+					// a4a port - ItemsDataViews
+					// @ts-expect-error -- this item.id assignation is to fix an issue with the DataViews component and item selection. It should be removed once the issue is fixed.
+					item.id = item.ID.toString();
```

Comment:
Is this still required? I tested removing it and I don't see any issue 🤔

Reply:
[Core still looks for .id](https://github.com/WordPress/gutenberg/blob/e7f0dc53ab3d8bbd53bd19129e00ed34aad67326/packages/dataviews/src/components/dataviews/index.tsx#L54), but I'm not sure if we need this on our end. Maybe it only matters if we're doing multi-row selection? We've sort of overloaded `selection` to signify what site is selected in the list view (and I presume only to apply a styling to it?)

Maybe that needs cleaning up? (e.g. css-target the row with something we pass in?)

Reply:
cc @okmttdhr

Reply:
Or was this an a4a need?

Reply:
Removing `item.id = item.ID.toString();` seems ok, getItemId is required. If we can't justify it probably should just remove?

Reply:
Yeah, I mean removing just that assignment, seems fine to remove

Reply:
Leaving for now, will follow up to remove this later.

Reply:
> Leaving for now, will follow up to remove this later.

Yeah! it's better; I'm not sure if it was already fixed in DataViews core.

Reply:
What's the issue with the core implementation of `getItemId`? If there's one, we should have a github issue to track it.

Reply:
Wasn't sure, ported from here: https://github.com/Automattic/wp-calypso/blob/aaad48c0ad0de9b63cf5d07a625101ec163263fd/client/a8c-for-agencies/components/items-dashboard/items-dataviews/index.tsx#L96

Reply:
I've checked the Gutenberg code, and I don't find it, so it seems it's already fixed. It seems safe to remove it and check that everything continues working. I remember that the issue was pretty evident.